lib/ecto/migrator.ex

```diff
@@ -451,6 +466,39 @@ defmodule Ecto.Migrator do
     |> collect_migrations(directories)
     |> Enum.sort_by(fn {_, version, _} -> version end)
   end
+  
+  use GenServer
+
+  @doc """
+  GenServer which starts, runs migrations and then shutdowns normally. 
+  Use this to run migrations as part of your application start.
+
+  ## Options
+
+    * `:repos` - Required option to tell the migrator which Repo's to 
+      migrate. Example: `repos: [MyApp.Repo]
+    * `:skip?` - Option to skip migrations.
+      Defaults to `false`.
+  """
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  @impl true
+  def init(opts) do
+    repos = Keyword.fetch!(opts, :repos)
+
+    skip? = Keyword.get(opts, :skip, false)
+    migrator = Keyword.get(opts, :migrator, &Ecto.Migrator.run/3)
+
+    unless skip? do
+      for repo <- repos do
+        {:ok, _, _} = with_repo(repo, &migrator.(&1, :up, all: true))
```

Comment:
Hey just a question 👋🏻

I don't know the context so I may be a bit off here but what about the other options one can pass to https://hexdocs.pm/ecto_sql/3.9.2/Ecto.Migrator.html#run/4 and https://hexdocs.pm/ecto_sql/3.9.2/Ecto.Migrator.html#up/4?

Seems like this feature is useful for releases and would be great to pass opts like `log_migrations_sql` or it could be necessary to pass `prefix` depending on the situation, and so on.

Reply:
You mean for documenting or passing those ops through to the migrator?

Reply:
Hi @jeregrine I mean being able to pass extra opts or documenting `:migrator` to make it public, I believe the later makes more sense.

So either pass extra opts:

```elixir
   {Ecto.Migrator,
      repos: Application.fetch_env!(:my_app, :ecto_repos),
      skip: System.get_env("SKIP_MIGRATIONS") == "true",
      log_migrations_sql: true # <- extra opts for migrator
    }
```

Or make the `:migrator` opt public:

```elixir
   {Ecto.Migrator,
      repos: Application.fetch_env!(:my_app, :ecto_repos),
      skip: System.get_env("SKIP_MIGRATIONS") == "true",
      migrator: fn repo -> Ecto.Migrator.run(repo, :up, all: true, log_migrations_sql: true) end
    }
```

For context, my use case requires passing `:prefix`.

Reply:
Sure!